### ISSUE TITLE
Saving caches and versions file even if some frameworks fail to build

### ISSUE THREAD

**Issue by ikesyo** [giginet/Scipio#153]
https://github.com/giginet/Scipio/blob/4c2cc399ab771993b2b94db6490fe9c926ee68aa/Sources/ScipioKit/Producer/FrameworkProducer.swift#L124-L141

Even if some frameworks fail to build, we can still save caches and version files for successfully-built frameworks. That would reduce build time for next attempts.

**Comment by mustafagunes** [comment]
Hi @ikesyo, I have been following and admiring your project for a while.

I have started using Scipio to reduce build times in the projects I am working on. I would like to contribute by solving this issue. Could you please assign it to me?

**Comment by ikesyo** [comment]
Thanks for the comment! I just assigned you. 😃

**Comment by mustafagunes** [comment]
should I create the development branch from the main branch?

**Comment by ikesyo** [comment]
Yes, please submit a pull request from a feature branch of your fork 🙏
